src/corecel/data/Copier.cc

```diff
@@ -11,9 +11,32 @@
 
 #include "corecel/device_runtime_api.h"
 #include "corecel/Macros.hh"
+#include "corecel/sys/Device.hh"
+#include "corecel/sys/Stream.hh"
 
 namespace celeritas
 {
+namespace
+{
+
+#if CELER_USE_DEVICE
+inline auto to_memcpy_kind(MemSpace src, MemSpace dst)
+{
+    if (src == MemSpace::host && dst == MemSpace::device)
+        return CELER_DEVICE_PREFIX(MemcpyHostToDevice);
+    else if (src == MemSpace::device && dst == MemSpace::host)
+        return CELER_DEVICE_PREFIX(MemcpyDeviceToHost);
+    else if (src == MemSpace::device && dst == MemSpace::device)
+        return CELER_DEVICE_PREFIX(MemcpyDeviceToDevice);
+    else
+        CELER_ASSERT_UNREACHABLE();
+    return CELER_DEVICE_PREFIX(MemcpyDefault);
```

Comment:
Surely both of these aren't needed 😅 I think I originally had the default there just to avoid a potentially uninitialized variable. Maybe just
```suggestion
    CELER_ASSERT_UNREACHABLE();
```

Reply:
yeah 😅 I remember having issues with unreachable and inline but it seems fine here.